crypt/cipher.go

```diff
@@ -284,11 +293,182 @@ func (c *cipher) decryptSegment(ciphertext string) (string, error) {
 	return string(plaintext), err
 }
 
+// Simple obfuscation routines
+
+func (c *cipher) obfuscateSegment(plaintext string) string {
+	if plaintext == "" {
+		return ""
+	}
+
+	// If the string isn't valid UTF8 then don't rotate; just
+	// prepend a 0.
+	if !utf8.ValidString(plaintext) {
+		return "0." + plaintext
+	}
+
+	// Calculate a simple rotation based on the filename and
+	// the nameKey
+	var dir int
+	for _, runeValue := range plaintext {
+		dir += int(runeValue)
+	}
+	dir = dir % 256
+
+	// We'll use this number to store in the result filename...
+	result := strconv.Itoa(dir) + "."
+
+	// but we'll augment it with the nameKey for real calculation
+	for i := 0; i < len(c.nameKey); i++ {
+		dir += int(c.nameKey[i])
+	}
+	dir = dir % 127
+
+	// Now for each character, depending on the range it is in
+	// we will actually rotate a different amount
+	var newChar string
+
+	for _, runeValue := range plaintext {
+		switch {
+		case string(runeValue) == obfuscQuote:
```

Comment:
Oh!  `'!'` gives a rune, whereas `"!"` gives a string?  Nice to know.  `perl` (my language of choice) doesn't distinguish :-)